peer/example/inboundpeer/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/btcsuite/btcd/addrmgr"
+	"github.com/btcsuite/btcd/peer"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btclog"
+)
+
+// newestSha returns the latest known block to this peer
+func newestSha() (sha *wire.ShaHash, height int64, err error) {
+	hash := "14a0810ac680a3eb3f82edc878cea25ec41d6b790744e5daeef"
+	height = 234439
+	sha, err = wire.NewShaHashFromStr(hash)
+	return
+}
+
+// lookupFunc returns the IP address corresponding to the given host string
+// It is used to initialize the address manager for testing purposes so
+// it is not implemented
+func lookupFunc(host string) ([]net.IP, error) {
+	return nil, errors.New("not implemented")
+}
+
+// handleVersionMsg is a peer listener which is called when the peer receives
+// a version message
+func handleVersionMsg(p *peer.Peer, msg *wire.MsgVersion) {
+	fmt.Printf("in handleVersionMsg\n")
+}
+
+func main() {
+	done := make(chan *peer.Peer)
+
+	addrMgr := addrmgr.New("test", lookupFunc)
+
+	logger, err := btclog.NewLoggerFromWriter(os.Stdout, btclog.DebugLvl)
+	if err != nil {
+		panic(err)
+	}
+	peer.UseLogger(logger)
+
+	peerCfg := &peer.Config{
+		Dial:             net.Dial,
+		UserAgentName:    "peer",
+		UserAgentVersion: "1.0",
+	}
+
+	listener, err := net.Listen("tcp4", "127.0.0.1:8333")
+	if err != nil {
+		panic(err)
+	}
+	conn, err := listener.Accept()
+	if err != nil {
+		panic(err)
+	}
+	p := peer.NewInboundPeer(newestSha, 0, done, wire.MainNet, addrMgr, peerCfg, conn)
+	err = p.Start()
+	if err != nil {
+		panic(err)
+	}
+
+	// Listeners can be added or removed to various message types
+	p.AddVersionMsgListener("handleVersionMsg", handleVersionMsg)
+
+	fmt.Printf("Started peer: %v\n", p)
+
+	time.AfterFunc(2*time.Minute, func() {
+		fmt.Printf("Shutdown peer: %v\n", p)
+		p.Shutdown()
+	})
+
+	<-done
```

Comment:
This might be a little more consistent as `p.WaitForShutdown()` much like the btcd server does and have the peer handle the channel portion rather than accepting one.